### DISCUSSION TITLE
Scale interface with egui-winit and egui-wgpu

### DISCUSSION THREAD

**Discussion by gadirom** [emilk/egui#3128] [Q&A]
I managed to render egui interface with GPU backend through egui-wgpu.
Also, what works is the offline rendering to a wgpu texture and showing it through an image.

The issue now is with the scaling of the interface. 

When I try to set `pixels_per_point` larger than 1.0 the interface scales, but then the mouse position becomes misaligned. 

I use `egui_winit.on_event` to handle events from a window, and it seems that egui-winit takes physical coords of the mouse as logical points.

This issue is both on native and web.

For web I tried to scale the contents as was advised here: https://github.com/emilk/egui/discussions/172

The interface indeed scales but then the layout goes off. E.g. I have a little panel above and an image below. With scaling the panel goes off the screen and becomes unreachable.

What are my options here? Should I continue trying to adapt egui-winit, or should I try some other integrations (if there are any)?

I'm interested primarily in web.

**Comment by 9SMTM6** [comment]
Is this maybe related to my issue here: https://github.com/emilk/egui/discussions/4660 ?

**Comment by emilk** [comment]
`eframe` does scaling correctly without any misalignment, and it uses `egui-winit` and `egui-wgpu`, so I suggest you either use `eframe` or look at the code for it and see what it does different.
